src/main/scala/com/wavesplatform/state2/diffs/TransactionDiffer.scala

```diff
@@ -24,7 +24,12 @@ object TransactionDiffer {
       t5 <- CommonValidation.disallowSendingGreaterThanBalance(s, settings, currentBlockTimestamp, t4)
       diff <- t5 match {
         case gtx: GenesisTransaction => GenesisTransactionDiff(currentBlockHeight)(gtx)
-        case ptx: PaymentTransaction => PaymentTransactionDiff(s, currentBlockHeight, settings, currentBlockTimestamp)(ptx)
+        case ptx: PaymentTransaction =>
+          if (currentBlockHeight <= settings.blockVersion3AfterHeight) {
+            PaymentTransactionDiff(s, currentBlockHeight, settings, currentBlockTimestamp)(ptx)
+          } else {
+            Left(UnsupportedTransactionType)
```

Comment:
I think `GenericError(s"Payment transaction is deprecated after h=${settings.blockVersion3AfterHeight}")`  would give more info to users, at least, it would be easier to find the exact line where `Left` appeared